### PR TITLE
change snap to terrain raycast modifier to take object position and t…

### PR DESCRIPTION
**Pull request by brnkhy**
…ile position into account

change snap to terrain modifier to take object position into account (due to center change options in mod stack)

### CODE REVIEW

File: sdkproject/Assets/Mapbox/Unity/MeshGeneration/Modifiers/MeshModifiers/SnapTerrainRaycastModifier.cs [hunk 1]
```diff
@@ -19,57 +21,31 @@ public override ModifierType Type
...
 
-			if (md.Vertices.Count > 0)
+			foreach (var sub in feature.Points)
 			{
-				for (int i = 0; i < md.Vertices.Count; i++)
+				for (int i = 0; i < sub.Count; i++)
 				{
-					var h = tile.QueryHeightData((float)((md.Vertices[i].x + tile.Rect.Size.x / 2) / tile.Rect.Size.x),
-						(float)((md.Vertices[i].z + tile.Rect.Size.y / 2) / tile.Rect.Size.y));
+					var h = tile.QueryHeightData((float)((sub[i].x + scaledX / 2) / scaledX), (float)((sub[i].z + scaledY / 2) / scaledY));
```
Comment (david-rhodes) [inline sdkproject/Assets/Mapbox/Unity/MeshGeneration/Modifiers/MeshModifiers/SnapTerrainRaycastModifier.cs]: @brnkhy This does not take into consideration the terrain height multiplier, does it?

**Comment by david-rhodes** [comment]
screen shot 2017-11-17 at 10 47 41 am
Upon closer inspection, it looks like all building inside same tile are set to the same height.

**Comment by brnkhy** [comment]
@david-rhodes I think you had that because you didn't had terrain collider on. Can you check it again please? It took me an hour to realize that as well but all worked properly when I enabled it.

File: sdkproject/Assets/Mapbox/Unity/MeshGeneration/Modifiers/MeshModifiers/SnapTerrainRaycastModifier.cs (hunk 1 above)
Comment (brnkhy) [reply sdkproject/Assets/Mapbox/Unity/MeshGeneration/Modifiers/MeshModifiers/SnapTerrainRaycastModifier.cs]: @david-rhodes it should I believe, can't see why not. multiplied height effects queried height anyway right?

**Comment by david-rhodes** [comment]
> I think you had that because you didn't had terrain collider on. Can you check it again please? It took me an hour to realize that as well but all worked properly when I enabled it.

The whole point of your implementation for fallback height was in case the raycast did not hit (no collider for terrain or hitting a tile edge). Therefore, you should test by turning off the terrain collider, right? 😏 At which point, it should use the basic height query.

**Comment by brnkhy** [comment]
@david-rhodes aah you meant that for fallback part, sorry totally missed that and you're right. 👍

**Comment by brnkhy** [comment]
@david-rhodes one problem with that though, it'll be hard to notice it didn't hit anything and using fallback and people will probably think it's raycasting while it actually is not. I'll also try to add an error notification there if it doesn't hit anything.

**Pull request merged**